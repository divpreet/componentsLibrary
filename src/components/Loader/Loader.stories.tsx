import * as React from 'react';
import { Story } from '@storybook/react';
import { boolean, number, text } from '@storybook/addon-knobs';
import Loader from './Loader';

export default {
    title: 'Components / Loader',
    component: Loader,
    componentSubtitle: 'Displays a Loader'
};

export const withKnobs: Story = () => (
    <Loader
        a11yId={text('Id', 'loader')}
        isLoading={boolean('Is Loading?', true)}
        size={number('Size', 50)}
    />
);
