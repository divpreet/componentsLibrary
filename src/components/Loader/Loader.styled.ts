import CircularProgress from '@material-ui/core/CircularProgress';
import { dclStyled } from '../Theme';

export const LoaderStyled = dclStyled(CircularProgress)`
    &.dclLoader {
        color: ${({ brandcolour }) => brandcolour};
    }
    .dclLoader__svg {
        width: ${({ size }) => size};
        height: ${({ size }) => size};
    }
`;
