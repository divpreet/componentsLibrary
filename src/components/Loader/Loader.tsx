import * as React from 'react';
import { LoaderStyled } from './Loader.styled';
import { useThemeContext } from '../Theme';
import * as sizes from '../../styles/dclSizes.json';

export interface LoaderProps {
    a11yId: string;
    className?: string;
    size?: number;
    type?: 'primary' | 'secondary';
    brandColour?: string;
    isLoading?: boolean;
}

const Loader: React.FC<LoaderProps> = ({
    a11yId,
    type = 'primary',
    brandColour,
    className,
    isLoading = true,
    size
}) => {
    if (!isLoading) {
        return null;
    }

    const { brandColour: dclBrandColour } = useThemeContext();
    return (
        <LoaderStyled
            className={className}
            classes={{
                root: 'dclLoader',
                svg: 'dclLoader__svg'
            }}
            brandcolour={type === 'primary' && (brandColour || dclBrandColour)}
            color={type}
            size={size ? `${size}px` : sizes.spacing.factor4.s6}
            id={a11yId}
        />
    );
};

export default Loader;
