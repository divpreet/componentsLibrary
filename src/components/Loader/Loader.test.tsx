import * as React from 'react';
import Loader from './Loader';
import { shallow } from 'enzyme';

describe('Loader', () => {
    let LoaderProps;
    beforeEach(() => {
        LoaderProps = {
            isLoading: true,
            a11yId: 'test'
        };
    });

    it('renders correctly', () => {
        const component = shallow(<Loader {...LoaderProps} />).dive();
        expect(component).toMatchSnapshot();
    });

    it('should show loader if isLoading is true', () => {
        const component = shallow(<Loader {...LoaderProps} />);

        expect(component).toBeDefined();
    });

    it('should hide loader if isLoading is false', () => {
        LoaderProps.isLoading = false;

        const component = shallow(<Loader {...LoaderProps} />);

        expect(component).toEqual({});
    });
});
