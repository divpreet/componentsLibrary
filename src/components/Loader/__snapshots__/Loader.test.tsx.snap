// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders correctly 1`] = `
.c0 .dclLoader__svg {
  width: 24px;
  height: 24px;
}

<WithStyles(ForwardRef(CircularProgress))
  className="c0"
  classes={
    Object {
      "root": "dclLoader",
      "svg": "dclLoader__svg",
    }
  }
  color="primary"
  id="test"
  size="24px"
  theme={
    Object {
      "alerts": Object {
        "button": "#32373F",
        "error": "#BD2030",
        "errorBg": "#F7E4E6",
        "info": "#0266C8",
        "infoBg": "#E0F3FE",
        "success": "#2A7D2A",
        "successBg": "#DFF6DF",
        "warning": "#F3AD2E",
        "warningBg": "#FDEED3",
      },
      "background": Object {
        "default": "#ffffff",
        "paper": "#F5F5F5",
        "tertiary": "#EFEFEB",
      },
      "brandColours": Object {
        "default": "#007DC3",
        "secondary": "#20908B",
      },
      "divider": "#d1d1d1",
      "global": Object {
        "headerBg": "#32373F",
        "navigationBg": "#32373F",
        "navigationSelectedBg": "#525965",
        "navigationText": "#ffffff",
      },
      "highlight": Object {
        "blue": "#E0F3FE",
        "green": "#DFF6DF",
        "red": "#F7E4E6",
        "yellow": "#FDEED3",
      },
      "meaningful": Object {
        "blue": "#0266C8",
        "green": "#2A7D2A",
        "purple": "#9D4098",
        "red": "#BD2030",
        "yellow": "#F3AD2E",
      },
      "primary": Object {
        "main": "#0266C8",
      },
      "text": Object {
        "disabled": "#707070",
        "primary": "#32373F",
        "secondary": "#525965",
      },
      "typography": Object {
        "body1": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "body2": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
        "h1": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "h2": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "h3": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "h4": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "h5": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "h6": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "htmlFontSize": 14,
        "subtitle1": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
        "subtitle2": Object {
          "fontFamily": "Roboto,-apple-system,BlinkMacSystemFont,Arial,Helvetica,sans-serif",
          "fontSize": "14px",
          "fontWeight": 400,
          "lineHeight": 1,
        },
      },
    }
  }
/>
`;
